### PR TITLE
Update mailer to use its own generated Messages i18n

### PR DESCRIPTION
**Pull request by i386**
@slide as per https://github.com/jenkinsci/mailer-plugin/pull/28 I've removed the usages of `build-helper-maven-plugin` and `maven-localizer-plugin`

### CODE REVIEW

File: src/main/java/jenkins/plugins/mailer/tasks/MimeMessageBuilder.java [hunk 1]
```diff
@@ -81,17 +81,15 @@
     private Set<InternetAddress> bcc = new LinkedHashSet<InternetAddress>();
 
     public MimeMessageBuilder() {
-        if (Jenkins.getInstance() != null) {
```
Comment (i386) [inline src/main/java/jenkins/plugins/mailer/tasks/MimeMessageBuilder.java]: Required to fix a find bugs issue preventing the build from passing

**Comment by slide** [comment]
Why do you think that the messages are not working as is? The package change should probably not be in this.

**Comment by i386** [comment]
@slide they are being pulled in by a Jenkins dependency thats was before the mailer split. The generated messages in this plugin are not being used at all. 

In #29 to get a newer toolchain I have bumped the parent POM and the built will fail. So I can either lump this change in with #29 or we can split it.

**Comment by slide** [comment]
Ok, understood. This gets a :+1: from me now. I'm not the maintainer though, not sure if there is one.

**Comment by i386** [comment]
@slide what do we do in this case?

**Comment by oleg-nenashev** [comment]
Core team releases it on demand.
@andresrc and @rsandell were ones who made previous releases, but actually anyone from jenkinsci-core can do it.

Well, actually it will require an upload permission patch

**Comment by slide** [comment]
Where is that permissions file stuff again? I may have done a release previously so I may have permissions. I can also merge the request if no one objects to anything.

**Comment by oleg-nenashev** [comment]
https://github.com/jenkins-infra/repository-permissions-updater

**Comment by slide** [comment]
Nope, I have not done a release.

**Comment by andresrc** [comment]
Let me take a look tomorrow and I can merge and perform the release. Thanks

**Comment by slide** [comment]
Does this change necessitate a release? I know @i386 has more changes that he is proposing.

**Comment by andresrc** [comment]
I mean, I'm available for checking and doing the release if needed for some other downstream change.

**Comment by i386** [comment]
Ive got more changes coming in #29

**Comment by i386** [comment]
@slide or anyone here have the authority to merge this one?

**Pull request merged**